doc/getting_started/build_sw.md

```diff
@@ -1,217 +1,336 @@
+### Linting Starlark 
 
-### Test Execution
+The OpenTitan supported linter for Bazel files is `buildifier`.
+It can be run with Bazel as shown below.
 
-Bazel will execute tests when invoked with `bazel test <label-expression>`.
-The test rules have various tags applied to them to allow for filtering of tests during dependency analysis.
-Our tests are typically tagged with verilator, cw310, or manual.
-The `--test_tag_filters` switch can be used to filter out tests that you either cannot or do not want to execute.
+Run the following to check if all `WORKSPACE`, `BUILD`, and `.bzl` files have been formatted correctly:
+```console
+bazel run //:buildifier_check
+```
+and run the following to fix them, if they are not formatted correctly.
+```console
+bazel run //:buildifier_fix
+```
 
-For example, if you do not have a CW310 FPGA board, you cannot execute the tests meant to execute on that board.
-You can instruct Bazel to not execute those tests:
+### Checking License Headers
 
+Lastly, the OpenTitan supported linter for checking that every source code file contains a license header may be run with:
 ```console
-bazel test --test_tag_filters=-cw310 //...
+bazel run //:license_check
 ```
 
-Rather than manually adding that command line switch to every invocation of bazel, you can add it to your `$HOME/.bazelrc` file:
+## Miscellaneous 
 
-```
-test --test_tag_filters=-cw310
+### Bazel-built Artifacts
+
+As decribed in the [OpenTitan Software]({{< relref "sw/" >}}) documentation, there are three categories of OpenTitan software, all of which are built with Bazel. These include:
+1. _device_ software,
+1. _OTBN_ software,
+1. _host_ software,
+
+Bazel produces various artifacts depending on the category of software that is built.
+
+#### Device Artifacts
+
+Device software is built and run on OpenTitan hardware.
+There are three OpenTitan "devices" for simulating/emulating OpenTitan hardware:
+1. DV simulation (i.e., RTL simulation with commercial simulators),
+1. Verilator simulation (i.e., RTL simulation with the open source Verilator simulator),
+1. FPGA.
+
+Different software artifacts are built depending on the OpenTitan device above.
+Specifically, building an executable `<target>` destined to run on the OpenTitan device `<device>` will output the following files under `bazel-out/`:
+* `<target>_<device>`: the linked program, in ELF format.
+* `<target>_<device>.bin`: the linked program, as a plain binary with ELF debug information removed.
+* `<target>_<device>.elf.s`: the disassembled program with inline source code.
+* `<target>_<device>.*.vmem`: a Verilog memory file which can be read by `$readmemh()` in Verilog code.
+
+Note, `<device>` will be in {`sim_dv`, `sim_verilator`, `fpga_cw310`}, and `<target>` will end in the suffix "`_prog`" for executable images destined for flash.
+
+#### OTBN Artifacts
+
+TODO: add documentation on OTBN artifacts.
```

Comment:
@jadephilipoom would you mind updating this part once this merges? I'm not as familiar with this as you are.

Reply:
Yep, I'll take care of it!

Reply:
Thanks!